internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/server/VeniceServerTest.java

```diff
@@ -128,4 +163,90 @@ public void testCheckBeforeJointClusterBeforeHelixInitializingCluster() throws E
       TestUtils.shutdownThread(serverAddingThread);
     }
   }
+
+  @Test
+  public void testMetadataFetchRequest() throws ExecutionException, InterruptedException, IOException {
+    Utils.thisIsLocalhost();
+    int servers = 6;
+    int replicationFactor = 2;
+
+    try (VeniceClusterWrapper cluster = ServiceFactory.getVeniceCluster(1, servers, 0, replicationFactor);
+        CloseableHttpAsyncClient client =
+            HttpClientUtils.getMinimalHttpClient(1, 1, Optional.of(SslUtils.getVeniceLocalSslFactory()));) {
+
+      HelixAdmin admin = new ZKHelixAdmin(cluster.getZk().getAddress());
+
+      HelixConfigScope configScope =
+          new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER).forCluster(cluster.getClusterName())
+              .build();
+      Map<String, String> clusterProperties = new HashMap<String, String>() {
+        {
+          put(ClusterConfig.ClusterConfigProperty.TOPOLOGY.name(), "/zone/instance");
+          put(ClusterConfig.ClusterConfigProperty.TOPOLOGY_AWARE_ENABLED.name(), "TRUE");
+          put(ClusterConfig.ClusterConfigProperty.FAULT_ZONE_TYPE.name(), "zone");
+        }
+      };
+
+      admin.setConfig(configScope, clusterProperties);
+
+      for (int i = 0; i < servers; i++) {
+        VeniceServerWrapper server = cluster.getVeniceServers().get(i);
+        String instanceName = server.getHost() + "_" + server.getPort();
+        String domain = "zone=zone_" + (char) (i % replicationFactor + 65) + ",instance=" + instanceName;
+
+        InstanceConfig instanceConfig = new InstanceConfig(instanceName);
+        instanceConfig.setDomain(domain);
+        instanceConfig.setHostName(server.getHost());
+        instanceConfig.setPort(String.valueOf(server.getPort()));
+
+        admin.setInstanceConfig(cluster.getClusterName(), instanceName, instanceConfig);
+      }
+
+      String storeName = cluster.createStore(1);
+
+      client.start();
+      HttpGet httpsRequest =
+          new HttpGet("http://" + cluster.getVeniceServers().get(0).getAddress() + "/metadata/" + storeName);
```

Comment:
If we only care one server's response, let's just create one server, instead of 6?

Reply:
True, I was testing if the helix groups are assigned properly but that shouldn't be part of the test for the endpoint.

Reply:
Alternatively, we can also test 6 servers' responses 😄

Reply:
Made the change to loop through the servers and test against all of them